metrics/operator_repo_stats.py

```diff
@@ -0,0 +1,83 @@
+"""Module for parsing operator repositories into useful metrics."""
+
+import time
+from pathlib import Path
+import subprocess
+from threading import Thread, Event
+
+import yaml
+from operator_repo import Repo
+
+from metrics.prometheus_metrics import (
+    MIGRATION_GAUGE,
+    BUNDLES_IN_REPO_GAUGE,
+    OPERATORS_IN_REPO_GAUGE,
+    MIGRATION_COUNT_GAUGE,
+)
+
+
+CLONE_DIR = Path("/tmp")  # nosec
+SYNC_DELAY = 86400  # 24 h in seconds
+
+
+def ensure_repo(repo: Path, git_url: str) -> None:
+    """Ensure a local repository exists and is up to date."""
+    if repo.exists():
+        pop = subprocess.Popen(
+            ["git", "pull"],
+            cwd=repo.absolute(),
+        )
+    else:
+        # pylint: disable=consider-using-with
```

Comment:
What is the reason for skipping a pylint here?

Reply:
Locally, I was getting a message from pylint that I should use the `with` statement for the `subprocess.Popen`. But I think it does not makes sense as there are two different branches of this function each using a different `subprocess.Popen` and just explicitly calling the wait method seems sufficient and cleaner in this scenario. (In my opinion)

Reply:
:+1: